File: adapter/code_java/adapter/SquarePegAdapter.java
package adapter;

import java.lang.Math;

import  round.RoundPeg;

import  square.SquarePeg;

public class SquarePegAdapter extends RoundPeg {

    private SquarePeg squarePeg;

    public SquarePegAdapter(SquarePeg squarePeg) {
        this.squarePeg = squarePeg;
    }

    public double getRaduis() {
        double result;
        result = (this.squarePeg.getWith() * Math.sqrt(2))/2;

        return result;
    }
}

File: 2-pattrons-de-structures/adapter/code-typescript/adapters/SquarePegAdapter.ts
import { RoundPeg } from "../round/RoundPeg.js";
import { SquarePeg } from "../square/SquarePeg.js";

export class SquarePegAdapter extends RoundPeg {
  constructor(private peg: SquarePeg) {
    super(peg.getWidth);
  }

  get getRaduis(): number {
    return (this.peg.getWidth * Math.sqrt(2)) / 2;
  }
}
